cmd/botkube-agent/main.go

```diff
@@ -353,13 +353,14 @@ func run(ctx context.Context) (err error) {
 		cfgReloader, err := reloader.Get(
 			remoteCfgEnabled,
 			logger.WithField(componentLogFieldKey, "Config Reloader"),
+			statusReporter,
 			deployClient,
 			dynamicCli,
 			restarter,
 			analyticsReporter,
 			*conf,
 			cfgVersion,
-			cfgManager,
+			cfgManager, statusReporter,
```

Comment:
`statusReporter` is passed in twice. Is this meant to be like that?

Reply:
the first one is obtained to report the status by `AckNewResourceVersion`.

While the last parameter is `resVerHolders ...ResourceVersionHolder` so it takes all the instances that share the `resourceVersion` in order to keep them in sync. So it's the same "Instance" but is consumed with different scopes (interfaces)

As the `statusReporter` is already a parameter, I can also handle that directly in the constructor, like:
```go
if remoteCfgEnabled {
		resVerHolders = append(resVerHolders, statusReporter)
```

The first one looks probably more like a Go approach where you define interfaces per usage, instead of having a "fat" one. But honestly, I don't have preferences. I can change it 👍